src/plugins/VersionRegistry.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.19;
+
+/// @title VersionRegistry for plugins
+/// @notice This contract serves as a registry for version information of various plugins.
+contract VersionRegistry {
+    struct Version {
+        uint8 major;
+        uint8 minor;
+        uint8 patch;
+    }
+
+    /// @dev Mapping from plugin address to its version information.
+    mapping(address => Version) private pluginVersions;
+    Version private latestVersion;
+
+    /// @dev Address of the contract owner or authorized entity for version management.
+    address private owner;
+
+    /// @notice Constructor sets the initial owner of the contract.
+    constructor() {
+        owner = msg.sender;
+    }
+
+    /// @notice Modifier to restrict certain functions to the contract owner only.
+    modifier onlyOwner() {
+        require(msg.sender == owner, "VersionRegistry: Caller is not the plugin owner");
+        _;
+    }
+
+    /// @notice Registers a plugin with its version information.
+    /// @dev This function can be called by the contract owner or authorized entity only.
+    /// @param plugin The address of the plugin to be registered.
+    /// @param major The major version number of the plugin.
+    /// @param minor The minor version number of the plugin.
+    /// @param patch The patch version number of the plugin.
+    function registerPlugin(address plugin, uint8 major, uint8 minor, uint8 patch) external onlyOwner {
+        Version memory newVersion = Version(major, minor, patch);
+
+        require(
+            (newVersion.major > latestVersion.major) ||
+            (newVersion.major == latestVersion.major && newVersion.minor > latestVersion.minor) ||
+            (newVersion.major == latestVersion.major && newVersion.minor == latestVersion.minor && newVersion.patch > latestVersion.patch),
+            "VersionRegistry: New version must be higher than the current latest version"
+        );
```

Comment:
Same as above, better to use if - revert than require.

Reply:
Resolved in b7d71190799bee8c720666466bfa71b4d68c5639)